legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts

```diff
@@ -486,7 +486,7 @@ export class ContractPublisher extends RPCConnectionHandler {
         : `0x${fetchedBytecode}`;
 
       const bytecodeHash = utils.solidityKeccak256(["bytes"], [bytecode]);
-      const contractId = predeployMetadata.name;
+      const contractId = extraMetadata.contractId || predeployMetadata.name;
```

Comment:
make sure there isnt any other places we use contractId - i know we use it as the key for a lot of things.

Also can you do the change in v5 as well?

Reply:
where can I find the v4 ContractPublisher class equivalent in v5?